src/lib/mutations.ts

```diff
@@ -13,13 +13,58 @@ export interface SuccessResponse<D> {
   error: null;
 }
 
+export interface FieldError {
+  field: string;
+  messages: string[];
+}
+
 export interface ErrorResponse<E> {
   data: null;
   error: E;
+  rawError: Record<string, unknown>;
+  fieldErrors: FieldError[];
 }
 
 type PromiseResult<D, E> = SuccessResponse<D> | ErrorResponse<E>;
 
+/**
+ * Extracts field errors from error response into a structured format
+ * @param error - The error response object
+ * @returns Array of field errors with field name and messages
+ */
+export function extractFieldErrors(error: any): FieldError[] {
+  const fieldErrors: FieldError[] = [];
+
+  if (!error) return fieldErrors;
+
+  // Handle nested errors structure like { errors: { field: [messages] } }
+  if (error.errors && typeof error.errors === 'object') {
+    Object.entries(error.errors).forEach(([key, value]) => {
+      // Case 1: Direct field errors (POST /v1/employments)
+      // { "errors": { "provisional_start_date": ["cannot be in a holiday"] } }
+      if (Array.isArray(value)) {
+        fieldErrors.push({
+          field: key,
+          messages: value.map((msg: any) => String(msg)),
+        });
+      }
+      // Case 2: Nested field errors (PATCH /v1/employments/*)
+      // { "errors": { "basic_information": { "provisional_start_date": ["cannot be in a holiday"] } } }
+      else if (typeof value === 'object' && value !== null) {
+        Object.entries(value).forEach(([nestedKey, nestedValue]) => {
+          if (Array.isArray(nestedValue)) {
+            fieldErrors.push({
+              field: nestedKey,
+              messages: nestedValue.map((msg: any) => String(msg)),
+            });
+          }
+        });
+      }
+    });
+  }
+  return fieldErrors;
+}
```

Comment:
suggestion:

I think you could make this a little simpler by returning early if errors is not an object. It would also get rid of the `else if`

```suggestion
export function extractFieldErrors(error: any): FieldError[] {
  if (!error || !error.errors || typeof error.errors !== 'object') return [];

  const fieldErrors: FieldError[] = [];

  Object.entries(error.errors).forEach(([key, value]) => {
    if (Array.isArray(value)) {
      fieldErrors.push({
        field: key,
        messages: value.map((msg: any) => String(msg)),
      });
      return;
    }

    if (value && typeof value === 'object') {
      Object.entries(value).forEach(([nestedKey, nestedValue]) => {
        if (Array.isArray(nestedValue)) {
          fieldErrors.push({
            field: nestedKey,
            messages: nestedValue.map((msg: any) => String(msg)),
          });
        }
      });
    }
  });

  return fieldErrors;
}
```